pkg/operators/machine-api-operator.go

```diff
@@ -75,4 +79,45 @@ var _ = Describe("[Feature:Operators] Machine API cluster operator status should
 		Expect(err).NotTo(HaveOccurred())
 		Expect(framework.IsStatusAvailable(client, "machine-api")).To(BeTrue())
 	})
+
+	It("be degraded when a pod owned by the operator is prevented from being available", func() {
+		c, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		// https://github.com/openshift/machine-api-operator/blob/d234cceb5de18b83aa0609d17db7d835f2d78973/pkg/operator/sync.go#L310-L313
+		mAPIControllersLabels := map[string]string{
+			"api":     "clusterapi",
+			"k8s-app": "controller",
+		}
+
+		Eventually(func() (bool, error) {
+			// get machine API controllers pods
+			podList := &v1.PodList{}
+			if err := c.List(context.TODO(), podList, client.MatchingLabels(mAPIControllersLabels)); err != nil {
+				klog.Errorf("failed to list pods: %v", err)
+				return false, nil
+			}
+			if len(podList.Items) < 1 {
+				klog.Errorf("list of pods is empty")
+				return false, nil
+			}
```

Comment:
Does this matter? If the list of pods is empty, status should be degraded right?

Reply:
I think it doesn't matter, but do we expect pods not to be running when at this part of the test?

Reply:
Well my thought was we go through this cycle, delete a pod, status isn't degraded yet, so start again, pod has not been recreated yet, could the status be degraded by this point? I am not sure, I would assume they are independent since they're different controllers controlling these actions